src/functions/indGraphSkinny.jl

```diff
@@ -0,0 +1,54 @@
+# This implements the variant when A is dense and m > n
+
+struct IndGraphSkinny{T <: RealOrComplex} <: IndGraph
+  m::Int
+  n::Int
+  A::AbstractArray{T,2}
+  AA::AbstractArray{T,2}
+  F::Base.LinAlg.Cholesky #LDL factorization
+end
+
+function IndGraphSkinny(A::AbstractArray{T,2}) where {T <: RealOrComplex}
+  m, n = size(A)
+  AA = A' * A
+
+  F = LinAlg.cholfact(AA + speye(n))
+  #normrows = vec(sqrt.(sum(abs2.(A), 2)))
+
+  IndGraphSkinny(m, n, A, AA, F)
+end
+
+function (f::IndGraphSkinny){T <: RealOrComplex}(x::AbstractArray{T,1}, y::AbstractArray{T, 1})
+  # the tolerance in the following line should be customizable
+  if norm(f.A * x - y, Inf) <= 1e-10
+    return 0.0
+  end
+  return +Inf
+end
+
+function prox!{T <: RealOrComplex}(
+    x::AbstractArray{T, 1},
+    y::AbstractArray{T, 1},
+    f::IndGraphSkinny,
+    c::AbstractArray{T, 1},
+    d::AbstractArray{T, 1})
+
+  x[:] = f.F \ (c + f.A' * d)
```

Comment:
I think you are raising a valid point, but I think case 1 should work fine. Even if m>>n it is just a single allocation of size m, the matrix is already dense m*n. But it's not an obvious choice.